ledfx/effects/water.py

```diff
@@ -0,0 +1,197 @@
+import queue
+
+import numpy as np
+import voluptuous as vol
+
+from ledfx.effects import smooth
+from ledfx.effects.audio import AudioReactiveEffect
+from ledfx.effects.hsv_effect import HSVEffect
+from ledfx.effects.math import triangle
+from ledfx.utils import empty_queue
+
+
+class Water(AudioReactiveEffect, HSVEffect):
+    """A rippling water effect.
+
+    Bass, mid, and high intensity levels create droplets evenly spaced along the
+    span.
+
+    References:
+        * https://mikro.naprvyraz.sk/docs/Coding/1/WATER.TXT
+        * https://github.com/Zygo/xscreensaver/blob/master/hacks/ripples.c
+    """
+
+    NAME = "Water"
+    CATEGORY = "Atmospherics"
+
+    CONFIG_SCHEMA = vol.Schema(
+        {
+            vol.Optional(
+                "speed",
+                description="Effect Speed modifier",
+                default=1,
+            ): vol.All(vol.Coerce(float), vol.Range(min=1, max=3)),
+            vol.Optional(
+                "vertical_shift",
+                description="Vertical Shift",
+                default=0.12,
+            ): vol.All(vol.Coerce(float), vol.Range(min=-0.2, max=1)),
+            vol.Optional(
+                "bass_size",
+                description="Size of bass ripples",
+                default=8,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0, max=15)),
+            vol.Optional(
+                "mids_size",
+                description="Size of mids ripples",
+                default=6,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0, max=15)),
+            vol.Optional(
+                "high_size",
+                description="Size of high ripples",
+                default=3,
+            ): vol.All(vol.Coerce(float), vol.Range(min=0, max=15)),
+            vol.Optional(
+                "viscosity",
+                description="Viscosity of ripples",
+                default=6,
+            ): vol.All(vol.Coerce(float), vol.Range(min=2, max=12)),
+        }
+    )
+
+    def __init__(self, ledfx, config):
+        super().__init__(ledfx, config)
+        # Queue will contain tuples of (pixel location, water height value)
+        self._drops_queue = queue.Queue()
+
+    def on_activate(self, pixel_count):
+        empty_queue(self._drops_queue)
+        # Double buffered rendering
+        self._buffer = np.zeros((2, pixel_count))
+        self._cur_buffer = 0
+        # Emitter positions are (percentage of span, speed and direction)
+        self._mids_emitters = [(0.25, 1.0), (0.75, -1.0)]
+        self._high_emitters = [
+            (0.125, 1.5),
+            (0.375, -2.5),
+            (0.625, 2.5),
+            (0.875, -1.5),
+        ]
+
+    def deactivate(self):
```

Comment:
Did you want to put the `empty_queue` here instead of on_activate, like you did in melt and sparkle?

Seems more useful on the deactivate?